internal/s3select/csv/args.go

```diff
@@ -112,6 +126,9 @@ func (args *ReaderArgs) UnmarshalXML(d *xml.Decoder, start xml.StartElement) (er
 						return fmt.Errorf("unsupported QuoteEscapeCharacter '%v'", s)
 					}
 				case "Comments":
+					if utf8.RuneCountInString(s) != 1 {
```

Comment:
This is not required. https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html
```suggestion
					if utf8.RuneCountInString(s) > 1 {
```

.. Should be good.

Reply:
@klauspost Seems like it would crash here:

> 		ret.Comment = []rune(args.CommentCharacter)[0]

in csv/reader.go

Reply:
Yeah, that is a problem. Not sure how/when we fill out defaults - `CommentCharacter` is one of the few with an actual documented default ;)